core/trino-main/src/main/java/io/trino/execution/buffer/OutputBufferMemoryManager.java

```diff
@@ -119,6 +132,13 @@ public void updateMemoryUsage(long bytesAdded)
         }
     }
 
+    private synchronized void recordOutputBufferUtilization() {
```

Comment:
This needs to be called from `OutputBuffer` before output buffer transitions to final state.

Reply:
done